### ISSUE TITLE
Don't always assume etcdctl is available

### ISSUE THREAD

**Issue by docwhat** [aespinosa/cookbook-flannel#2]
Currently, the template hardcodes the existence of both a config and an `etcdctl` executable.

Instead, it should be possible to disable this behavior.

Use cases:

* If I want to use another resource or mechanism for setting the configuration.
* I only have `etcdctl` on certain hosts (e.g. the masters in a kubernetes cluster).

PS: This works great on CentOS 7. I think adding a `supports` would be acceptable!

**Comment by aespinosa** [comment]
Thanks for the suggestion. Probably the cleanest implementation would look like a separate `action :configure` that does only the configuration via etcd and completely remote the ExectStartPre.

How does systemd deal with empty ExectStartPre= directives? You can probably do some in-recipe monkey-patching like this:


```
flannel.flannel_service 'default'

def flannel.etcdctl_command; end
```

And that would create the template without them. Just thinking how a minor release can look like to incorporate this. If there's not much demand for such a backport. I'll just work on the idea I mentioned earlier and create a major point release as it breaks the old `:start` action.

**Comment by docwhat** [comment]
Exposing the template cookbook as a variable would work too.

I did the monkey patch at the class level with `/bin/true`.

**Comment by aespinosa** [comment]
Released as 1.0.0 

https://supermarket.chef.io/cookbooks/flannel/versions/1.0.0
